Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py

```diff
@@ -51,52 +52,66 @@
 #####################################
 #  Open Communication with the QOP  #
 #####################################
-qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+# qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+qmm = QuantumMachinesManager()
 
-qm = qmm.open_qm(config)
+#######################
+# Simulate or execute #
+#######################
 
-job = qm.execute(T1)
-res_handles = job.result_handles
-I_handle = res_handles.get("I")
-Q_handle = res_handles.get("Q")
-iteration_handle = res_handles.get("iteration")
-I_handle.wait_for_values(1)
-Q_handle.wait_for_values(1)
-iteration_handle.wait_for_values(1)
-next_percent = 0.1  # First time print 10%
+simulate = True
 
+if simulate:
+    simulation_config = SimulationConfig(duration=1000)  # in clock cycles
+    job = qmm.simulate(config, T1, simulation_config)
+    job.get_simulated_samples().con1.plot()
 
-def on_close(event):
-    event.canvas.stop_event_loop()
-    job.halt()
+else:
 
+    qm = qmm.open_qm(config)
 
-f = plt.figure()
-f.canvas.mpl_connect("close_event", on_close)
-print("Progress =", end=" ")
+    job = qm.execute(T1)
+    res_handles = job.result_handles
+    I_handle = res_handles.get("I")
+    Q_handle = res_handles.get("Q")
+    iteration_handle = res_handles.get("iteration")
+    I_handle.wait_for_values(1)
+    Q_handle.wait_for_values(1)
+    iteration_handle.wait_for_values(1)
+    next_percent = 0.1  # First time print 10%
+
+
+    def on_close(event):
+        event.canvas.stop_event_loop()
+        job.halt()
+
+
+    f = plt.figure()
+    f.canvas.mpl_connect("close_event", on_close)
```

Comment:
While this tool is not available in the py qua tools, I would put this function in the configuration file to avoid duplicates.
I am using this function:
```
# Live plotting
def interrupt_on_close(figure, current_job):
    def on_close(event):
        print("Execution stopped by user!")
        current_job.halt()
        event.canvas.stop_event_loop()

    figure.canvas.mpl_connect("close_event", on_close)
```
The function can be called using 
```
fig = plt.figure()
interrupt_on_close(fig, job)  #  Interrupts the job when closing the figure
```

Reply:
nice